src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -262,10 +262,12 @@ await subscriber.Value.InitializeSubscriberAsync(this, _activeConfiguredProjectS
 
                     foreach (var subscriber in Subscribers)
                     {
-                        subscriber.Value.AddSubscriptionsAsync(newProjectContext);
+                        Shell.ThreadHelper.JoinableTaskFactory.Run(() => subscriber.Value.AddSubscriptionsAsync(newProjectContext));
```

Comment:
I have objectiins to doing in 15.3 unless we're solving an actual existing bug. This becomes easy to introduce other issues and other bugs such as deadlocks on shutdown.